app/ember/src/client/preview/types.ts

```diff
@@ -14,3 +17,8 @@ export interface OptionsArgs {
   context: any;
   element: any;
 }
+
+export interface StoryFnEmberReturnType<Args = DefaultArgs> {
+  template: ReturnType<typeof hbs>;
+  context: Args;
+}
```

Comment:
Unlike react, ember stories just return a `{ template, context }` object, not an "element"

https://github.com/storybookjs/storybook/blob/c174d7429c2593e051cc2dd1361bb28bff71f75b/app/react/src/client/preview/types.ts

Reply:
`context` can be any pojo. It is not `Args` only. More like: `Args & Record<string, unknown>` ?

Reply:
`Args = DefaultArgs` which is already an indexible type. So it can be anything

https://github.com/storybookjs/storybook/blob/e940b85e5eef9eeed37402bf387e4f80d897a08c/lib/addons/src/types.ts#L35-L37

Reply:
not necessarily, if you pass in another type `StoryFnEmberReturnType<Foo>` and if `Foo` is an interface with defined keys, then it does not work. I think it must be something like this:

```ts
export interface StoryFnEmberReturnType<Args extends DefaultArgs = DefaultArgs> {
  // ... 
```